src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -247,23 +307,33 @@ getCommentContent :: Node -> Maybe Text
 getCommentContent node = do
   txt <- getHTMLText node
   T.stripSuffix "-->" =<< T.stripPrefix "<!--" (T.strip txt)
-  where
-    getHTMLText :: Node -> Maybe Text
-    getHTMLText (Node _ (HTML_BLOCK txt) _) = Just txt
-    getHTMLText (Node _ (HTML_INLINE txt) _) = Just txt
-    getHTMLText _ = Nothing
+
+getHTMLText :: Node -> Maybe Text
+getHTMLText (Node _ (HTML_BLOCK txt) _) = Just txt
+getHTMLText (Node _ (HTML_INLINE txt) _) = Just txt
+getHTMLText _ = Nothing
 
 getXrefcheckContent :: Node -> Maybe Text
 getXrefcheckContent node =
   let notStripped = T.stripPrefix "xrefcheck:" . T.strip =<<
         getCommentContent node
   in T.strip <$> notStripped
 
+-- | Get correct possition of an annotation node. There is a bug in
+-- `commonmarkToNode` from the `cmark-gfm` package that affects one line
+-- `HTML_BLOCK` nodes, those node have wrong end line and end column positions.
+-- As our annotations are always oneliners, we can fix this by simply setting
+-- end line equals to start line and calculating end column from start column
+-- and annotation length.
```

Comment:
Can you please open an issue in their repo? Preferably with a small reproducible example, like something they could paste into their GHCI.

Reply:
[Done](https://github.com/kivikakk/cmark-gfm-hs/issues/23)